### PR TITLE
Update script codes to verify UniswapV3 contracts by using blockscout and some 

### PR DESCRIPTION
**Pull request by AaronLee22**
- Updated verify-predeploy-contract-hardhat.sh and hardhat.config.ts in order to verify  UniswapV3 contracts by using blockscout.
- Updated some code for initialize function and constructor function for UniswapV3 contracts.

### CODE REVIEW

**Comment by AaronLee22** [comment]
On my way of fixing  e2e test.

**Comment by 0x6e616d** [comment]
@AaronLee22 should convert this PR to DRAFT to prevent triggering CI every commit

**Comment by AaronLee22** [comment]
> @AaronLee22 should convert this PR to DRAFT to prevent triggering CI every commit. We should optimize the CI workflow in the next sprint

Oh, I see let me convert it to draft. Thank you!

**Comment by nguyenzung** [comment]
@AaronLee22 I think we can run e2e test locally, if it works, we can push to remote branch

File: packages/tokamak/contracts-bedrock/tsconfig.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "include": [
     "deploy-config/**/*",
     "deploy-config/**/*.json",
-    "scripts/**/*"
+    "scripts/**/*",
+    "**/*.ts"
```
Comment (rlgns98kr) [inline packages/tokamak/contracts-bedrock/tsconfig.json:10]: Should we include all file in here ? 🙇🏼‍♂️

File: packages/tokamak/contracts-bedrock/hardhat.config.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      chainId: 901,
+    },
+    'thanos-sepolia-test': {
+      url: "https://explorer.thanos-sepolia-test.tokamak.network",
+      chainId: 111551118080,
+    }
+  },
+  etherscan: {
+    apiKey: "YOUR_API_KEY",
+    customChains: [
```
Comment (rlgns98kr) [inline packages/tokamak/contracts-bedrock/hardhat.config.ts:19]: I think we should add thanos-sepolia-test in the customChains as well to be able to verify the network. Can you check it?

File: op-chain-ops/genesis/setters.go [hunk 3]
```diff
@@ -70,6 +77,7 @@ func setProxies(db vm.StateDB, proxyAdminAddr common.Address, namespace *big.Int
 		case predeploys.FiatTokenV2_2Addr:
 			db.SetCode(addr, fiatTokenProxyBytecode)
 			db.SetState(addr, AdminSlotForZepplin, eth.AddressAsLeftPaddedHash(proxyAdminAddr))
+		case predeploys.NonfungibleTokenPositionDescriptorAddr:
```
Comment (nguyenzung) [inline op-chain-ops/genesis/setters.go:80]: Hi @AaronLee22 
Correct me if I am wrong, NonfungibleTokenPositionDescriptorAddr is not a proxy contract?

Comment (AaronLee22) [reply op-chain-ops/genesis/setters.go:80]: Oh, you are right it was actually an implementation contract for transparentUpgradableProxy. let me fix this part! Thank you!

File: packages/tokamak/contracts-bedrock/hardhat.config.ts (hunk 2 above)
Comment (AaronLee22) [reply packages/tokamak/contracts-bedrock/hardhat.config.ts:19]: Sure, let me check it out!

**Review by xxeonge — COMMENTED** [review]
Hi @AaronLee22
Correct me if I am wrong.😄 As I understand it, NonfungibleTokenPositionDescriptor's proxy is TransparentUpgradeableProxy, right?
If that's correct then you probably need to modify the IsProxied function !
When I ran devnet-up, it seems that the implementation address was not set. Thank you!

**Comment by rlgns98kr** [comment]
Congratulations!
I will review quickly! 😍❤️

**Comment by AaronLee22** [comment]
> Congratulations! I will review quickly! 😍❤️

Thank you Steven!!

**Review by rlgns98kr — APPROVED** [review]
Thank you very much!

**Pull request closed without merging**